OutOfSchool/OutOfSchool.DataAccess/Repository/WorkshopRepository.cs

```diff
@@ -97,15 +97,25 @@ public override async Task<Workshop> Create(Workshop workshop)
         return await Task.FromResult(workshop).ConfigureAwait(false);
     }
 
-    public Task<List<Workshop>> GetAllWithDeleted(Expression<Func<Workshop, bool>> whereExpression)
+    public Task<List<Workshop>> GetAllWithDeleted(DateTime updatedAfter, int from, int size)
     {
-        IQueryable<Workshop> query = db.Workshops;
+        IQueryable<Workshop> query = dbSet;
 
-        if (whereExpression != null)
-        {
-            query = query.Where(whereExpression);
-        }
+        query = updatedAfter == default
+            ? query.Where(workshop => !workshop.IsDeleted)
+            : query.Where(workshop => workshop.UpdatedAt > updatedAfter || workshop.DeleteDate > updatedAfter);
+
+        return query.Skip(from).Take(size).ToListAsync();
+    }
+
+    public Task<int> CountWithDeleted(DateTime updatedAfter)
```

Comment:
Same name vs logic issue. `WithDeleted` in name but `!workshop.IsDeleted` in logic